src/image_processing/process_node.cpp

```diff
@@ -0,0 +1,68 @@
+#include <cv_bridge/cv_bridge.h>
+#include <image_transport/image_transport.h>
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/image_encodings.h>
+
+#include <opencv/cv.h>
+#include "robosub/VisualTarget.h"
+
+using cv::Mat;
+
+ros::Publisher pub;
+image_transport::Subscriber sub;
+
+//process img and return center as target
+robosub::VisualTarget process(Mat &img) {
+  // img is in BGR
+  // Do any processing and place the result in img
+  cvtColor(img, img, CV_BGR2HSV);
+  cvtColor(img, img, CV_HSV2BGR);
+
+  // TODO: hard-coded for now, will change later
+  robosub::VisualTarget target;
+
+  target.x = 20;
+  target.y = 60;
+
+  return target;
+}
+
+// call process on img to target
+void callback(const sensor_msgs::ImageConstPtr &msg) {
+  cv_bridge::CvImagePtr cv_ptr;
+  try {
+    cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+  } catch (cv_bridge::Exception &e) {
+    ROS_ERROR("cv_bridge exception: %s", e.what());
+    return;
+  }
+
+  robosub::VisualTarget target_ctr;
+  target_ctr = process(cv_ptr->image);
+  pub.publish(target_ctr);
+}
+
+//subscribe and publish target msg
+int main(int argc, char **argv) {
+  std::string input, target;
+
+  ros::init(argc, argv, "image_processor");
+
+  if (!ros::param::get("input", input)) {
```

Comment:
I think we should replace these params with remaps. So just subscribe directly to the "input" topic and put a remap in the launch file